src/audio/volume/volume.c

```diff
@@ -426,10 +544,21 @@ static struct comp_dev *volume_new(const struct comp_driver *drv,
  */
 static void volume_free(struct comp_dev *dev)
 {
+#if CONFIG_IPC_MAJOR_4
+	struct vol_data *cd = comp_get_drvdata(dev);
+	struct ipc4_peak_volume_regs regs;
+#else
 	struct vol_data *cd = comp_get_drvdata(dev);
+#endif
 
 	comp_dbg(dev, "volume_free()");
 
+#if CONFIG_IPC_MAJOR_4
+	/* clear mailbox */
```

Comment:
and i'd move the declaration of regs inside this #if so there'd be only one #if/#endif